components/Card.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import Link from 'next/link';
+import TextTruncate from 'react-text-truncate';
+interface CardProps {
+  title: string;
+  body: string;
+  icon?: string;
+  link?: string;
+  image?: string;
+  extended?: boolean;
+  headerSize?: 'small' | 'medium' | 'large';
+  bodyTextSize?: 'small' | 'medium' | 'large';
+}
+
+const CardBody = ({
+  title,
+  body,
+  icon,
+  link,
+  image,
+  extended,
+  headerSize,
+  bodyTextSize,
+}: CardProps) => {
+  const headerSizeClasses: Record<string, string> = {
+    small: 'text-[0.9rem]',
+    medium: 'text-[1.3rem]',
+    large: 'text-[2rem]',
+  };
+  const bodyTextSizeClasses: Record<string, string> = {
+    small: 'text-[0.85rem]',
+    medium: 'text-[1rem]',
+    large: 'text-[1.5rem]',
+  };
+  return (
+    <div className='group relative h-full w-full rounded-lg border border-gray-200 bg-white p-6 px-12 shadow-3xl dark:shadow-2xl dark:shadow-slate-900 transition-colors ease-in-out hover:bg-slate-100 dark:bg-slate-800 hover:dark:bg-slate-900/30'>
+      <div className='flex justify-center '>
+        {image && <img src={image} className='h-32 p-2' />}
```

Comment:
I think it is better to use Next.js prebuilt `Image` Component. It does some optimization under the hood.

https://nextjs.org/learn-pages-router/seo/improve/images
https://nextjs.org/docs/pages/building-your-application/optimizing/images